staging/src/k8s.io/kube-aggregator/pkg/cmd/server/start.go

```diff
@@ -142,6 +142,9 @@ func (o AggregatorOptions) RunAggregator(stopCh <-chan struct{}) error {
 	)
 	serverConfig.OpenAPIConfig = genericapiserver.DefaultOpenAPIConfig(openapi.GetOpenAPIDefinitions, openapinamer.NewDefinitionNamer(aggregatorscheme.Scheme))
 	serverConfig.OpenAPIConfig.Info.Title = "kube-aggregator"
+	// prevent generic API server from installing the OpenAPI handler. Aggregator server
+	// has its own customized OpenAPI handler.
+	genericConfig.SkipOpenAPIInstallation = true
```

Comment:
oh, you were right, this should also be set in `createAggregatorConfig`. I missed that the aggregator config is inheriting the `OpenAPIConfig` field from `kubeAPIServerConfig` there.